### PR TITLE
Fix crash when splash screen is not specified for FlutterFragmentActivity

### PR DESCRIPTION
**Pull request by jiahaog**
When the manifest file doesn't specify `io.flutter.embedding.android.SplashScreenDrawable` and `FlutterFragmentActivity` is used, the app crashes with a `android.content.res.Resources$NotFoundException: Resource ID #0x0` exception.

This should have been part of flutter/engine#13660. 

- Fix the splash screen tests in `FlutterFragmentActivityTest` as well – they should have been instantiating a `FlutterFragmentActivity` instead of `FlutterActivity`.

## Related Issue

- flutter/flutter#85292 - users may want to disable the custom splash screen in order to take advantage of the preDraw listener
- Fixes flutter/flutter#44242

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by jiahaog** [comment]
@blasten any thoughts on how this should be tested? Seems like flutter/flutter#44242 was filed previously for the tests when #13660 was created.

**Review by blasten — COMMENTED** [review]
change LG, but could we add a test?

**Comment by jiahaog** [comment]
@blasten sorry for the delay, I have added the tests

File: shell/platform/android/test/io/flutter/embedding/android/FlutterActivityTest.java [hunk 1]
```diff
@@ -391,6 +391,26 @@ public void itLoadsThemedSplashScreenDrawable() throws PackageManager.NameNotFou
     assertNotNull(splashScreen);
   }
 
+  @Test
+  public void itWithMetadataWithoutSplashScreenResourceDoesNotLoadsSplashScreenDrawable()
```
Comment (blasten) [inline shell/platform/android/test/io/flutter/embedding/android/FlutterActivityTest.java:395]: is this test name correct?

File: shell/platform/android/test/io/flutter/embedding/android/FlutterFragmentActivityTest.java [hunk 2]
```diff
@@ -271,26 +271,46 @@ public void itLoadsThemedSplashScreenDrawable() throws PackageManager.NameNotFou
...
     shadowOf(RuntimeEnvironment.application.getPackageManager()).addOrUpdateActivity(activityInfo);
 
     // It should load the drawable.
-    SplashScreen splashScreen = flutterActivity.provideSplashScreen();
+    SplashScreen splashScreen = activity.provideSplashScreen();
     assertNotNull(splashScreen);
   }
 
+  @Test
+  public void itWithMetadataWithoutSplashScreenResourceDoesNotLoadsSplashScreenDrawable()
```
Comment (blasten) [inline shell/platform/android/test/io/flutter/embedding/android/FlutterFragmentActivityTest.java:295]: ditto

**Review by blasten — APPROVED** [review]
LGTM + comment about test name

File: shell/platform/android/test/io/flutter/embedding/android/FlutterActivityTest.java (hunk 1 above)
Comment (jiahaog) [reply shell/platform/android/test/io/flutter/embedding/android/FlutterActivityTest.java]: More or less, it has the metadata file but does not have the lines 
```
activityInfo.metaData.putInt(
        FlutterActivityLaunchConfigs.SPLASH_SCREEN_META_DATA_KEY,
        SplashShadowResources.THEMED_SPLASH_DRAWABLE_ID);
```

to put the `<meta-data>`  entry. Updated it to make it a bit clearer though.

File: shell/platform/android/test/io/flutter/embedding/android/FlutterFragmentActivityTest.java (hunk 2 above)
Comment (jiahaog) [reply shell/platform/android/test/io/flutter/embedding/android/FlutterFragmentActivityTest.java]: Same as above

**Pull request merged**